### PR TITLE
Fix/dataprovider test refactor

### PR DESCRIPTION
**Pull request by violetteavi**
Code refactor to follow best practices
- [x] Don't use new/delete use smart pointers, preferably unique_ptr (like we do in the reset of the code). 
- [x] Prefer ctor/dtor rathere than SetUp/TearDown, as documented in the gtest primer
- [x] Reseting gflags is as easy as writing to them (at the end of the day they are global variables): FLAGS_noimages_rectified = false.
- [x] No malloc or other C black magic, you should not need to use it, and if you find yourself using it, reconsider the design of the code.
- [x] Refactor \#define macro closer to the code

### CODE REVIEW

**Comment by violetteavi** [comment]
@Toni Rosinol Code compiles and tests run fine, looks like there's some weirdness with the Euroc performance scripts. Do you recognize this?
Log:
```
 python3.6 /root/Kimera-VIO-Evaluation/evaluation/main_evaluation.py -r -a -v --save_plots --save_boxplots --save_results --write_website /root/Kimera-VIO-Evaluation/experiments/jenkins_euroc.yaml
I0420 23:40:19.690479 1270 font_manager.py:1337] generated new fontManager

  0%|          | 0/11 [00:00<?, ?it/s]I0420 23:40:22.927937 1270 evaluation_lib.py:298] Run dataset: V1_01_easy
ERROR: unknown command line flag 'log_euroc_gt_data'
E0420 23:40:23.132654 1270 evaluation_lib.py:163] Failed pipeline run!
I0420 23:40:23.132884 1270 evaluation_lib.py:301] [91m Dataset: V1_01_easy failed!! [00m

  0%|          | 0/11 [00:00<?, ?it/s]
[92m Run pipeline: Euroc[00m
[92m Finished evaluation for dataset: V1_01_easy[00m
Traceback (most recent call last):
  File "/root/Kimera-VIO-Evaluation/evaluation/main_evaluation.py", line 72, in <module>
    if run(args):
  File "/root/Kimera-VIO-Evaluation/evaluation/main_evaluation.py", line 22, in run
    dataset_evaluator.evaluate()
  File "/root/Kimera-VIO-Evaluation/evaluation/evaluation_lib.py", line 302, in evaluate
    raise Exception("Failed to run dataset %s." % dataset['name'])
Exception: Failed to run dataset V1_01_easy.

```

**Comment by ToniRV** [comment]
@violetteavi the problem is with this gflag:
```
ERROR: unknown command line flag 'log_euroc_gt_data'
```
Did we remove it?

**Comment by ToniRV** [comment]
Would be great to merge this asap :)

File: tests/testDataProviderModule.cpp [hunk 1]
```diff
@@ -27,92 +27,46 @@
...
-#define TEST_TIMEOUT_SUCCESS_END(X)                              \
-  finished.set_value(true);                                      \
-  }, std::ref(promisedFinished)).detach();           \
-  ASSERT_EQ(futureResult.wait_for(std::chrono::milliseconds(X)), \
-            std::future_status::timeout);
-#endif
 static constexpr int default_timeout = 100000;
 static constexpr int no_timeout = std::numeric_limits<int>::max();
 
+DECLARE_bool(images_rectified);
```
Comment (ToniRV) [inline tests/testDataProviderModule.cpp:33]: Fix:
```
DECLARE_bool(images_rectified, true, "Add comment on what this does.");
```
(Second arg is default value)

File: tests/testDataProviderModule.cpp [hunk 2]
```diff
@@ -27,92 +27,46 @@
...
+
 /* ************************************************************************** */
 // Testing data
 static const double tol = 1e-7;
 
 class TestDataProviderModule : public ::testing::Test {
```
Comment (ToniRV) [inline tests/testDataProviderModule.cpp:39]: namespace everything under VIO, as always.

**Comment by ToniRV** [comment]
@andrewhaeffner feel free to merge (just two comments above) :+1:

**Comment by violetteavi** [comment]
@ToniRV Great! I, alas, am not authorized to merge this pull request-- would you mind?

**Pull request merged**